SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -29,54 +29,61 @@ using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 template <typename DataT>
 using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
 
-// Calls simd constructor in provided invocation context, which depends on the
-// TestCaseT type. TestCaseT is a struct, that should have call_simd_ctor method
-// that should return constructed object of simd class.
-// This function returns std::vector instance with the output data.
-template <typename DataT, int NumElems, typename TestCaseT>
-auto call_simd(sycl::queue &queue, const shared_vector<DataT> &ref_data) {
-
-  shared_vector<DataT> result{NumElems, shared_allocator<DataT>{queue}};
-
-  queue.submit([&](sycl::handler &cgh) {
-    const auto ref = ref_data.data();
-    auto out = result.data();
+template <typename DataT, int NumElems = 0>
+class TestDescription : public ITestDescription {
+protected:
+  const std::string m_data_type;
```

Comment:
```suggestion
class TestDescription : public ITestDescription {
  const std::string m_data_type;
```

Are you going to have inheritance further more and to access these elements in child? Looks like it would result in duplication of the `to_string()` logic. I assume I could suggest something  - but we would need a specific example to discuss :)

Reply:
I don't know how, but I expect that `m_data_type` and other fields would be re-used in classes-inheritances
But they are `protected` and don't accessible by `.` or `->` operators

I don't have a specific example, all my examples - it's just imaginations :)
If it necessary I can make this fields are `private`

Reply:
Let's start with the `private` for now

Reply:
I've done it in this commit: a46c51bc932553cab7ea460e9442943784a0eaf4